### PR TITLE
Controller

### PR DESCRIPTION
**Pull request by shaiTheKimhi**

### CODE REVIEW

File: Modules/Control/Control.py [hunk 1]
```diff
@@ -9,24 +13,37 @@
...
-        self.magnetometer = magnetometerMT()
+        self.magnetometer = magnetometer_factory.initialize()
+        self.gyro = gyro_factory.initialize()
+        self.magnetorquer = magnetorquerMT.magnetorquerMT()
         self.time = time.time()
-        #TODO : Add a secondary thread that changes calls setControl
-        
-
+        # secondary thread that changes calls setControl
+        thread = Thread(target = set_control ,args = (2,))
```
Comment (yotam180) [inline Modules/Control/Control.py]: The comma is in the wrong place...

File: Modules/Control/Control.py [hunk 2]
```diff
@@ -9,24 +13,37 @@
...
+        while(True):
+            arr = setControl(mode = mode)
+            self.magnetorquer.set_x(arr[0])
+            self.magnetorquer.set_y(arr[1])
+            self.magnetorquer.set_z(arr[2])
+            time.sleep(config.TIME_DIFFERENCE)
     #TODO: Add offset for meneuvering to certain locations on earth
     def set_control(self, mode=1, offset=[0,0]):
-        #makes the direction change for the measured depolar magnetic field (cos function value)
+        # makes
```
Comment (yotam180) [inline Modules/Control/Control.py]: Did you just leave the first word of the comment?

File: Modules/Control/Control.py [hunk 3]
```diff
@@ -9,24 +13,37 @@
...
         if(mode == 1):
             field = self.magnetometer.get_axes()
             timer = (self.time-time.time()) % (90*60)
```
Comment (yotam180) [inline Modules/Control/Control.py]: Classic. Operation spacing.

**Review by yotam180 — CHANGES_REQUESTED** [review]
The comments left below AND APPLY OPERATOR SPACING.

File: Modules/Control/Control.py [hunk 4]
```diff
@@ -9,24 +13,36 @@
...
             z = (field[1]*torque[0] - filed[0]*torque[1]) / (field[2]*torque[1] - field[1]*torque[2])
             y = (-field[0] - z * field[2]) / field[1]
-            return np.array([1,y,z])
-        #makes the direction change for circular magnetic field, in order to go by the field
+            arr = np.ndarray([1,y,z])
+            tilt = np.ndarray([self.gyro.accel()])
+            return arr - tilt
         elif(mode == 2):
-            return np.array([1,1,1])
+            #makes the direction change for circular magnetic field, in order to go by the field
```
Comment (yotam180) [inline Modules/Control/Control.py]: Capitalize comments and add a space between `#` and `Makes`

**Review by yotam180 — CHANGES_REQUESTED** [review]
Operator spacing is still a massive problem in your pull requests.

**Review by yotam180 — CHANGES_REQUESTED** [review]
Please just give some space to your operators already :slightly_frowning_face::gun:

**Comment by shaiTheKimhi** [comment]
I added operator spacing, could you be specific and not so general???

**Comment by shaiTheKimhi** [comment]
I already added operator spacing in the places you requested, could you be specific with your requests and not so general ???

**Comment by yotam180** [comment]
No. I am tired of writing "Operator spcaing" every line again and again. The file is still full of unspaced operators. Hunting them down is your job since you are the one who brought them to life.

File: Modules/Control/Control.py [hunk 5]
```diff
@@ -35,14 +35,14 @@ def set_control(self, mode=1, offset=[0,0]):
             field = self.magnetometer.get_axes()
             timer = (self.time-time.time()) % (90 * 60)
             torque = get_angle_vector(get_angle(timer))
-            z = (field[1]*torque[0] - filed[0]*torque[1]) / (field[2]*torque[1] - field[1]*torque[2])
+            z = (field[1]*torque[0] - filed[0]*torque[1]) / (field[2] * torque[1] - field[1] * torque[2])
```
Comment (yotam180) [inline Modules/Control/Control.py]: > "Whom do you show half work to?" - Wise elder Orna Segal

File: Modules/Control/Control.py [hunk 6]
```diff
@@ -85,12 +85,12 @@ def get_angle_vector(angle:int)->numpy.ndarray:
         x = 1
         value = 1 / (math.cos(angle) ** 2)
         z = math.sqrt(value-x)
```
Comment (yotam180) [inline Modules/Control/Control.py]: .

File: Modules/Control/Control.py [hunk 7]
```diff
@@ -85,12 +85,12 @@ def get_angle_vector(angle:int)->numpy.ndarray:
...
     else:
         if(angle == 90):
-            return numpy.array([0,0,1])
+            return numpy.array([0, 0, 1])
         else:
-            return numpy.array([0,0,-1])
+            return numpy.array([0, 0, -1])
 
 #This function gets the satellite location (angle) by the measured  magnetic field
 def get_angle_by_field(field:numpy.ndarray ,mode=1)->float:
```
Comment (yotam180) [inline Modules/Control/Control.py]: .

File: Modules/Control/Control.py [hunk 8]
```diff
@@ -104,6 +104,6 @@ def get_angle_by_field(field:numpy.ndarray ,mode=1)->float:
         b = field[1]
         y = math.sqrt(1 / (a**2 + b**2) )
```
Comment (yotam180) [inline Modules/Control/Control.py]: .

File: Modules/Control/Control.py [hunk 9]
```diff
@@ -104,6 +104,6 @@ def get_angle_by_field(field:numpy.ndarray ,mode=1)->float:
...
         x = math.sqrt(a**2 / (a**2 * (a**2 + y**2))
-        temp = math.sqrt(x**2/y**2 + 1)
+        temp = math.sqrt(x**2 / y**2 + 1)
```
Comment (yotam180) [inline Modules/Control/Control.py]: .

File: Modules/Control/Control.py [hunk 10]
```diff
@@ -104,6 +104,6 @@ def get_angle_by_field(field:numpy.ndarray ,mode=1)->float:
         b = field[1]
         y = math.sqrt(1 / (a**2 + b**2) )
         x = math.sqrt(a**2 / (a**2 * (a**2 + y**2))
```
Comment (yotam180) [inline Modules/Control/Control.py]: .

**Review by yotam180 — CHANGES_REQUESTED** [review]
This is starting to look better.

File: Modules/Control/Control.py (hunk 5 above)
Comment (shaiTheKimhi) [reply Modules/Control/Control.py]: It is a quote that orna was mistaken in

Comment (yotam180) [reply Modules/Control/Control.py]: Well still you have spaced only a half of the line xx

**Comment by shaiTheKimhi** [comment]
Are there any more changes you requested?
Because I did add operator spacing in all the places you had requested.

**Review by yotam180 — APPROVED** [review]
I think it's good now. Thanks for your time 😄

**Comment by shaiTheKimhi** [comment]
Can one of you merge this pull request please?

**Comment by yotam180** [comment]
Sure

**Comment by shaiTheKimhi** [comment]
Thank you.

**Pull request merged**